### PR TITLE
Get image blob

### PR DESCRIPTION
**Pull request by sahanDissanayake**

### CODE REVIEW

**Comment by hu9o** [comment]
You are right in that ``getImage()`` is not satisfying in its current state.

However I have to disagree with this PR : ``getImageBlob()`` is mostly a copy of ``getImage()``, and the two functions achieve almost the exact same thing.

They differ only at the point where they begin to be too specific : **``getImage()`` should simply return the canvas element**, and let the user get a data URL, a blob, display it directly, or whatever.

**Comment by sahanDissanayake** [comment]
Sweet, Can we return just the canvas then ?

**Comment by hu9o** [comment]
That's exactly what I would recommend !
Although for the best, it would still mean changing the public API...

@mosch, what do you say ?

**Comment by sahanDissanayake** [comment]
yeah thats the thing, i did not wanted to introduce any breaking changes. what if we pass a var to the getImage() and if nothing is passed we return dataURL but if the variable == 'canvas' or 'blob' we return  it accordingly ? 

so no breaking changes

**Comment by hu9o** [comment]
Other breaking changes have just been introduced, so I made a PR : #98

**Comment by mosch** [comment]
imho we dont need this one since we have #98 right?

File: dist/index.js [hunk 1]
```diff
@@ -156,6 +156,26 @@
...
+
+            dom.width = dimensions.width;
+            dom.height = dimensions.height;
+
+            context.globalCompositeOperation = 'destination-over';
+
+            this.paintImage(context, this.state.image, border);
+
+            dom.toBlob(function(blob){
+                return blob;
```
Comment (hu9o) [inline dist/index.js]: this cannot work, BTW

**Comment by hu9o** [comment]
Right.

For the record, one can get a blob by doing :
```javascript
var canvas = editor.getImage();
canvas.toBlob(function(blob){
        // use `blob` here
    }, type, quality);
``` #

**Comment by sahanDissanayake** [comment]
Thanks!!

File: dist/index.js (hunk 1 above)
Comment (mosch) [inline dist/index.js]: whops, looks like a quirky git merge

**Comment by mosch** [comment]
so this is done!

**Pull request closed without merging**